examples/demo-rollup/src/test_rpc.rs

```diff
@@ -152,30 +207,347 @@ fn test_get_batches() {
+                        TestExpect{
+                        data:
+                        format!(r#"{{"jsonrpc":"2.0","method":"ledger_getTransactions","params":[[{random_tx_num}], "Full"],"id":1}}"#),
+                        expected:
+                        format!(r#"{{"jsonrpc":"2.0","result":[{tx_formatted}],"id":1}}"#)},
+                        ]
+                        , slots);
+
+                    return Ok(());
+                }
+
+                curr_tx_num += batch.tx_receipts.len();
+            }
+        }
+
+        let data = format!(r#"{{"jsonrpc":"2.0","method":"ledger_getTransactions","params":[[{random_tx_num}]],"id":1}}"#);
+        let expected : String = r#"{"jsonrpc":"2.0","result":[null],"id":1}"#.to_string();
+        test_helper(vec![TestExpect{data, expected}], slots);
+
+    }
+
+    #[test]
+    fn proptest_get_events((slots, tx_id_to_event_range, _total_num_batches) in arb_slots(10, 10), random_event_num in 1..10000){
+        let mut curr_tx_num = 1;
+
+        let random_event_num_usize = usize::try_from(random_event_num).unwrap();
+
+        for slot in &slots{
+            for batch in slot.batch_receipts(){
+                for tx in &batch.tx_receipts{
+                    let (start_event_range, end_event_range) = tx_id_to_event_range.get(&curr_tx_num).unwrap();
+                    if *start_event_range > random_event_num_usize {
+                        break;
+                    }
+
+                    if random_event_num_usize < *end_event_range {
+                        let event_index = random_event_num_usize - *start_event_range;
+                        let event : &Event = tx.events.get(event_index).unwrap();
+
+                        let key_str_vec : Vec<String> = event.key().inner().iter().map(|x| x.to_string()).collect();
+                        let key_str = key_str_vec.join(",");
+
+                        let value_str_vec : Vec<String> = event.value().inner().iter().map(|x| x.to_string()).collect();
+                        let value_str = value_str_vec.join(",");
+
+                        let event_formatted = format!(r#"{{"key":[{key_str}],"value":[{value_str}]}}"#);
+
+
+                    test_helper(vec![TestExpect{
+                        data:
+                        format!(r#"{{"jsonrpc":"2.0","method":"ledger_getEvents","params":[{random_event_num_usize}],"id":1}}"#),
+                        expected:
+                        format!(r#"{{"jsonrpc":"2.0","result":[{event_formatted}],"id":1}}"#)},
+                        ]
+                        , slots);
+
+                    return Ok(());
+                    }
+                    curr_tx_num += 1;
+                }
+
+            }
+
+        }
+
+        let data = format!(r#"{{"jsonrpc":"2.0","method":"ledger_getEvents","params":[{random_event_num}],"id":1}}"#);
+        let expected : String= r#"{"jsonrpc":"2.0","result":[null],"id":1}"#.to_string();
```

Comment:
```suggestion
        let expected: String = r#"{"jsonrpc":"2.0","result":[null],"id":1}"#.to_string();
```

Reply:
It looks like formatting inside macro isn't working...

Reply:
There's no way around this, sadly. We could use attribute macros (https://docs.rs/proptest-attr-macro/latest/proptest_attr_macro/ or https://github.com/proptest-rs/proptest/issues/153 once it's released) but it's not worth it IMO.